marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/ConstraintsSpec.hs

```diff
@@ -0,0 +1,178 @@
+{-# LANGUAGE DataKinds #-}
+
+module Language.Marlowe.Runtime.Transaction.ConstraintsSpec
+  ( spec
+  ) where
+
+import qualified Cardano.Api as C
+import qualified Cardano.Api.Shelley as C
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.Map as Map
+import Data.Monoid (First(..), getFirst)
+import Data.Ratio ((%))
+import qualified Data.Set as Set
+import Gen.Cardano.Api.Typed
+import Language.Marlowe.Runtime.Cardano.Api (fromCardanoAddressAny, fromCardanoAddressInEra)
+import qualified Language.Marlowe.Runtime.ChainSync.Api as Chain
+import qualified Language.Marlowe.Runtime.Core.Api as Core.Api
+import Language.Marlowe.Runtime.Transaction.Constraints (TxConstraints(..))
+import qualified Language.Marlowe.Runtime.Transaction.Constraints as Constraints
+import qualified Language.Marlowe.Runtime.Transaction.Constraints as TxConstraints
+import Test.Hspec (Spec, expectationFailure, shouldBe)
+import qualified Test.Hspec as Hspec
+import qualified Test.Hspec.QuickCheck as Hspec.QuickCheck
+import qualified Test.QuickCheck as QuickCheck
+import Test.QuickCheck.Hedgehog (hedgehog)
+
+byteStringGen :: QuickCheck.Gen ByteString
+byteStringGen = BS.pack <$> QuickCheck.arbitrary
+
+emptyTxConstraints :: TxConstraints 'Core.Api.V1
+emptyTxConstraints = TxConstraints
```

Comment:
This isn't necessary to define - `TxConstraints` is an instance of `Monoid` so `mempty` does the same thing